### ISSUE TITLE
Can't connect to server

### ISSUE THREAD

**Issue by Shen** [gary-kim/riotchat#113]
I tried to ad a matrix-server and get an error-page. Here are some details:
![nextcloud-element-error](https://user-images.githubusercontent.com/81589/89461732-aa310300-d76c-11ea-9531-553601e758a1.jpg)

Running Nextcloud 17.0.3.1

**Comment by gary-kim** [comment]
Is your Matrix homeserver set up for SSO by any chance?

**Comment by Shen** [comment]
Yes, it is!

**Comment by gary-kim** [comment]
Okay, currently, SSO homeservers can only be connected to if it is set as the default homeserver in the admin settings. I've been a bit hesitant to make it work for all homeservers as it does represent a very slight security risk but maybe it should be fine.

**Comment by Shen** [comment]
It is set as the default homeserver in the admin settings.

**Comment by gary-kim** [comment]
That's strange, #80 should have fixed that issue. Some further investigation may be required

**Comment by Shen** [comment]
You got a mail with some more details. Maybe it helps.

**Comment by paulrbr-fl** [comment]
:wave: I posted this on the [#elementfornextcloud-general:garykim.dev](https://matrix.to/#/!CDWVdagHmbnQcqIPyv:garykim.dev?via=garykim.dev&via=matrix.org&via=kde.org) support channel but I think it could be useful for a later bugfix here too.

I managed to have a SSO enabled matrix homeserver working by:

- Open the element webapp from the `/apps/riotchat/riot/` nextcloud path (I.e. `https://your-nextcloud.org/apps/riotchat/riot/`) in a new tab (so element is loaded with no iframe constraint)
- Login with SSO flow as usual (I only needed to whitelist my nextcloud domain in Synapse with the `sso:client_whitelist:` server yaml config)
- Close the tab once logged in
- Open the riotchat nextcloud app (at `/apps/riotchat/` path) and “It works!” 🎉

So maybe it would be nice to have a configuration option in the nextcloud app to display a button to login in a popup for those kind of setups. The difficult part might be to detect in the popup tab when the login flow is finished, to close the tab and reload the nextcloud app

**Comment by paulRbr** [comment]
:wave: I posted this on the [#elementfornextcloud-general:garykim.dev](https://matrix.to/#/!CDWVdagHmbnQcqIPyv:garykim.dev?via=garykim.dev&via=matrix.org&via=kde.org) support channel but I think it could be useful for a later bugfix here too.

I managed to have a SSO enabled matrix homeserver working by:

- Open the element webapp from the `/apps/riotchat/riot/` nextcloud path (I.e. `https://your-nextcloud.org/apps/riotchat/riot/`) in a new tab (so element is loaded with no iframe constraint)
- Login with SSO flow as usual (I only needed to whitelist my nextcloud domain in Synapse with the `sso:client_whitelist:` server yaml config)
- Close the tab once logged in
- Open the riotchat nextcloud app (at `/apps/riotchat/` path) and “It works!” 🎉

So maybe it would be nice to have a configuration option in the nextcloud app to display a button to login in a popup for those kind of setups. The difficult part might be to detect in the popup tab when the login flow is finished, to close the tab and reload the nextcloud app
